app/src/main/java/com/amaze/filemanager/filesystem/FileUtil.java

```diff
@@ -334,11 +349,19 @@ public static final void writeUriToStorage(@NonNull final MainActivity mainActiv
                         }
                     }
                 }
-                return null;
+                return (retval.size() > 0) ? retval : null;
             }
 
             @Override
             public Void onPostExecute(Object result) {
+                if(result !=  null) {
+                    List<String> paths = (List<String>) result;
+                    if (paths.size() == 1) {
```

Comment:
Why are you returning a `List` and only using the first element?

Reply:
It is possible that more than one file can be saved per `FileUtil.writeUriToStorage()` since it accepts `List<Uri>`.

If more than one `Uri` can be saved, I'll want to make the Toast pops up as
 - if one `Uri` is saved, display the exact path of the location of the file saved
 - if more than one `Uri` are saved, simply notify user how many files are saved instead of popping up multiple Toasts